text/0122-sdk-hub-scope-merge.md

```diff
@@ -0,0 +1,161 @@
+
+A scope has a reference to a client. By default it will reference a noop client. You can bind a client to a scope via `scope.setClient()`. 
+The client is inherited by forked scopes.
+
+```js
+const client1 = new Client();
+const scope = new Scope();
+
+scope.getClient(); // <-- returns a noop client by default
+
+scope.setClient(client1);
+scope.getClient(); // <-- returns client1
+```
+
+The current scope may be kept similar to how we currently keep the current hub, but this is SDK specific and not part of this RFC.
+
+When calling `getScope()` before a scope was made the current one (=before init was called), we will return a scope for a noop client. 
+A noop client is a regular client that simply does not send anything.
+
+This way, the API for `getClient()` can always return a client, and users do not have to guard against this being undefined all the time. 
+We may also expose a util like `sentryIsInitialized()` that checks if the current client is a Noop client (which currently you could have checked as `getCurrentHub().getClient() === undefined`).
+
+If you want to have multiple isolated clients, you can achieve this easily with this new setup:
+
+```js
+const client1 = new Client();
+const client2 = new Client();
+
+const scope1 = new Scope();
+const scope2 = new Scope();
+
+scope1.setClient(client1);
+scope2.setClient(client2);
+
+scope1.captureException(); // <-- isolated from scope2
+```
+
+## Scopes
+
+Scopes behave similar to how they behave today. 
+When a scope is forked via `withScope()`, a new scope is created that inherits all data currently set on the parent scope.
+
+The main change to Scopes is that they do not push/pop anymore, but instead fork an execution context (in languages where this makes sense/is possible).
+Basically, `withScope()` should behave like `hub.run()` does today in languages that have execution context forking.
+
+`client.getScope()` should return the current scope of this client in the current execution context.
+
+From a users perspective, this should mostly not be noticeable - they can always run `getScope()` to get the current scope, or `withScope(callback)` to fork a new scope off the current scope.
+
+You can make a scope the current one via `setCurrentScope(scope)`, which should bind the scope to the current execution context (or a global, in SDKs without execution context). This is a replacement for the current APIs like `makeMain(hub)` or `setCurrentHub(hub)`.
+
+You can still clone scopes manually the same way as before, e.g. via `Scope.clone(oldScope)` or a similar API. In contrast to `withScope()`, this will _not_ fork an execution context.
+
+You can update the client of a scope via `scope.setClient(newClient)`. This will not affect any scope that has already been forked off this scope, but any scope forked off _after_ the client was updated will also receive the updated client.
+
+## Should users care about Clients?
```

Comment:
I added this section as it came up a few times in PR feedback. The idea is to make it explicit that the client should _not_ be a main user-facing concept, but more a power user feature.